kubeflow/trainer/rhai/speculator.py

```
@@ -0,0 +1,1735 @@
+    if "_start_data_progress_server" in globals():
+        _start_data_progress_server(hidden_states_dir, num_samples)  # noqa: F821
+
+    if "_set_phase" in globals():
+        _set_phase("extracting", 15)  # noqa: F821
+
+    print(
+        f"[Kubeflow] Extracting hidden states from '{verifier_model}' to '{hidden_states_dir}'",
+        flush=True,
+    )
+
+    script_path = "/tmp/data_generation_offline.py"
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/data_generation_offline.py"

(S108)